src/pages/resultsView/ResultsViewURLWrapper.ts

```diff
@@ -389,18 +407,39 @@ export default class ResultsViewURLWrapper
         );
     }
 
-    @computed public get comparisonSubTabId() {
-        return this.query.comparison_subtab || GroupComparisonTab.OVERLAP;
-    }
-
     @autobind
     public setTabId(tabId: ResultsViewTab, replace?: boolean) {
         this.updateURL({}, `results/${tabId}`, false, replace);
     }
 
+    public get subTab() {
+        const tabSegment = this.getCurrentTabSegment();
+        if (!tabSegment) {
+            return '';
+        }
+
+        // Extract subtab name after "results/tabSegment/" in the URL path
+        const subtabMatch = this.pathName.match(
+            new RegExp(`\/results\/${tabSegment}\/([^\/?#]+)`)
+        );
+        return subtabMatch?.[1] || '';
+    }
+
     @autobind
-    public setComparisonSubTabId(tabId: GroupComparisonTab) {
-        this.updateURL({ comparison_subtab: tabId });
+    public setSubTab(
+        subtab: ResultsViewComparisonSubTab | ResultsViewPathwaysSubTab
+    ) {
+        const tabSegment = this.getCurrentTabSegment();
+        if (tabSegment) {
+            const sanitized = sanitizeForUrl(subtab);
+            this.updateURL({}, `results/${tabSegment}/${sanitized}`);
+        }
+    }
+
+    private getCurrentTabSegment(): string | undefined {
+        // Extract tab name after "/results/" in the URL path
+        const match = this.pathName.match(/\/results\/([^\/]+)/);
+        return match?.[1];
     }
```

Comment:
We should probably add a pure utility function outside of the class, something like below, so that we can add unit tests.

```JavaScript
function getCurrentTabSegment(pathName: string): string | undefined {
    // Extract tab name after "/results/" in the URL path
    const match = pathName.match(/\/results\/([^\/]+)/);
    return match?.[1];
}
```

Then we can modify this one as a `get` function

```suggestion
    public get currentTabSegment(): string | undefined {
        return getCurrentTabSegment(this.pathName);
    }
```

Reply:
Hi @onursumer 
I looked into adding a utility function as suggested, and discovered we already have this functionality in ResultsViewPageHelpers via the getTabId() function, which is already unit tested.

I have updated the implementation to reuse this existing function instead of creating a duplicate


```
private get currentTabSegment(): string | undefined {
  return getTabId(this.pathName);
}
```

Let me know any adjustments for this approach.

Reply:
Amazing!